code/game/machinery/atmoalter/clamp.dm

```diff
@@ -80,39 +69,29 @@
 
 	qdel(target.parent)
 
-	if(network_node1)
-		qdel(network_node1)
-	if(network_node2)
-		qdel(network_node2)
-
-	var/obj/machinery/atmospherics/pipe/node1 = null
-	var/obj/machinery/atmospherics/pipe/node2 = null
+	for(var/datum/pipe_network/netnode in network_nodes)
+		qdel(netnode)
+	network_nodes = null
 
-	if(target.node1)
-		target.node1.build_network()
-		node1 = target.node1
-	if(target.node2)
-		target.node2.build_network()
-		node2 = target.node2
-	if(istype(node1) && node1.parent)
-		var/datum/pipeline/P1 = node1.parent
-		P1.build_pipeline(node1)
-		qdel(P1)
-	if(istype(node2) && node2.parent)
-		var/datum/pipeline/P2 = node2.parent
-		P2.build_pipeline(node2)
-		qdel(P2)
-//  P1.build_network()
-//  P2.build_network()
+	for(var/obj/machinery/atmospherics/pipe/node in target.pipeline_expansion())
+		node.build_network()
+		LAZYADD(network_nodes, node)
+		var/datum/pipeline/P = node.parent
+		P.build_pipeline(node)
+		qdel(P)
```

Comment:
Is this some hack to distribute air around? I don't get it. Why are we deleting P here?